dora/tests/src/test/java/alluxio/client/rest/S3ObjectTest.java

```diff
@@ -0,0 +1,412 @@
+   */
+  @Test
+  public void getNonExistentObject() throws Exception {
+    final String bucket = "bucket";
+    final String objectKey = "bucket/object";
+    createBucketTestCase(bucket).checkResponseCode(Status.OK.getStatusCode());
+    getTestCase(objectKey).checkResponseCode(Status.NOT_FOUND.getStatusCode())
+        .checkErrorCode(S3ErrorCode.Name.NO_SUCH_KEY);
+  }
+
+  /**
+   * Gets a non-existent directory.
+   */
+  @Test
+  public void getNonExistentDirectory() throws Exception {
+    final String bucket = "bucket";
+    final String objectKey = "bucket/object/";
+    createBucketTestCase(bucket).checkResponseCode(Status.OK.getStatusCode());
+    getTestCase(objectKey).checkResponseCode(Status.NOT_FOUND.getStatusCode())
+        .checkErrorCode(S3ErrorCode.Name.NO_SUCH_KEY);
+  }
+
+  /**
+   * Puts a small object and gets it.
+   */
+  @Test
+  public void putAndGetSmallObject() throws Exception {
+    final String bucket = "bucket";
+    final String objectKey = "bucket/object";
+    final byte[] object = "Hello World!".getBytes();
+
+    createBucketTestCase(bucket).checkResponseCode(Status.OK.getStatusCode());
+    createObjectTestCase(objectKey, object).checkResponseCode(Status.OK.getStatusCode());
+    getTestCase(objectKey).checkResponseCode(Status.OK.getStatusCode()).checkResponse(object);
+  }
+
+  /**
+   * Puts a large object and gets it.
+   */
+  @Ignore
+  public void putAndGetLargeObject() throws Exception {
+    // TODO(Xinran Dong): error occurs when getting large object.
+  }
+
+  /**
+   * Puts a directory and heads it.
+   */
+  @Test
+  public void putAndHeadDirectory() throws Exception {
+    final String bucket = "bucket";
+    final String folderKey = bucket + AlluxioURI.SEPARATOR + "folder" + AlluxioURI.SEPARATOR;
+
+    headTestCase(bucket).checkResponseCode(Status.NOT_FOUND.getStatusCode());
+    createBucketTestCase(bucket).checkResponseCode(Status.OK.getStatusCode());
+    createObjectTestCase(folderKey, EMPTY_OBJECT).checkResponseCode(Status.OK.getStatusCode());
+    headTestCase(folderKey).checkResponseCode(Status.OK.getStatusCode());
+  }
+
+  /**
+   * Puts and heads directories.
+   */
+  @Test
+  public void putAndHeadDirectories() throws Exception {
+    final String bucket = "bucket" + AlluxioURI.SEPARATOR;
+    final String objectKey1 = bucket + "folder0" + AlluxioURI.SEPARATOR;
+    final String objectKey2 = objectKey1 + "folder1" + AlluxioURI.SEPARATOR;
+    createBucketTestCase(bucket).checkResponseCode(Status.OK.getStatusCode());
+    createObjectTestCase(objectKey2, EMPTY_OBJECT).checkResponseCode(Status.OK.getStatusCode());
+    headTestCase(objectKey1).checkResponseCode(Status.OK.getStatusCode());
+    headTestCase(objectKey2).checkResponseCode(Status.OK.getStatusCode());
+  }
+
+  /**
+   * Overwrites an existent object.
+   */
+  @Test
+  public void overwriteObject() throws Exception {
+    final String bucket = "bucket";
+    final String objectKey = "bucket/object";
```

Comment:
the objectKey is the object name, and then construct the full path according to the bucket name and object name. follow the previous suggestion